src/Meilisearch/Query/MainQueryBuilder.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\Meilisearch\Query;
+
+use Meilisearch\Contracts\SearchQuery;
+
+final class MainQueryBuilder implements MainQueryBuilderInterface
```

Comment:
I feel like having both a main query builder and a sub query builder is a bit weird, but I can't put my finger on it... What's your feeling about it?

Reply:
Tbh, I don't feel this way 😄 Main builder is with no doubts different than a subquery, needs different data to be built, and is built in a different way. If yous still feel this awkwardness, we can iterate over it after the PR is merge, wdyt?